Lesson31/Kadir/my-app/src/components/AddButton/index.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from "react";
+import "./styles.css";
+
+export const AddButton = ({ key, data, setData}) => {
+  const [text, setText] = useState("");
+  const [newItem, setNewItem] = useState(); // Define a state to store the newItem
+
+  const handleChange = (event) => {
+    setText(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    text.trim() !== "" &&
+      setNewItem({
+        key: key + 1,
+        id: Math.floor(Math.random() * 1000),
+        title: text,
+        completed: false,
+      });
+    setText(""); // Clear the input field
+  };
+
+  const handleAddItem = (newItem) => {
```

Comment:
This method is not setting data correctly and actually not needed at all. Only `handleChange` and `handleSubmit` are used.